src/overlays/actors/ovl_Object_Kankyo/z_object_kankyo.c

```diff
@@ -33,42 +38,689 @@ const ActorInit Object_Kankyo_InitVars = {
+}
+
+void func_808DC454(ObjectKankyo* this, GlobalContext* globalCtx) {
+    s16 i;
+    u32 tempI;
+    f32 phi_f20;
+    f32 spD0;
+    f32 spCC;
+    f32 spC8;
+    f32 spC4;
+    f32 spC0;
+    f32 spBC;
+    f32 temp_f0_4;
+    f32 temp_f22;
+    f32 temp_f24;
+    f32 temp_f28;
+    f32 x = globalCtx->view.at.x - globalCtx->view.eye.x;
+    f32 y = globalCtx->view.at.y - globalCtx->view.eye.y;
+    f32 z = globalCtx->view.at.z - globalCtx->view.eye.z;
+    f32 sqrt = sqrtf(SQ(x) + SQ(y) + SQ(z));
+    f32 temp_120 = 120.0f;
+    f32 temp_f30;
+    Vec3f sp88;
+    s32 pad;
+
+    spD0 = x / sqrt;
+    spCC = y / sqrt;
+    spC8 = z / sqrt;
+
+    for (i = 0; i < globalCtx->envCtx.unk_F2[2]; i++) {
+        switch (this->unk_14C[i].unk_1C) {
+            case 0:
+                this->unk_14C[i].unk_00 = globalCtx->view.eye.x + (spD0 * 120.0f);
+                this->unk_14C[i].unk_04 = globalCtx->view.eye.y + (spCC * 120.0f);
+                this->unk_14C[i].unk_08 = globalCtx->view.eye.z + (spC8 * 120.0f);
+                this->unk_14C[i].unk_0C = (Rand_ZeroOne() - 0.5f) * (2.0f * temp_120);
+
+                temp_f22 = (func_800DFCB4(GET_ACTIVE_CAM(globalCtx)) * 0.004f) + 60.0f;
+                if (temp_f22 < 20.0f) {
+                    temp_f22 = 20.0f;
+                }
+
+                if (this->unk_114E == 0) {
+                    this->unk_14C[i].unk_10 = temp_f22;
+                } else {
+                    this->unk_14C[i].unk_10 += temp_f22;
+                    tempI = i;
+                    if (globalCtx->envCtx.unk_F2[2] == (tempI + 1)) {
+                        this->unk_114E = 0;
+                    }
+                }
+
+                this->unk_14C[i].unk_14 = (Rand_ZeroOne() - 0.5f) * (2.0f * temp_120);
+                if (globalCtx->envCtx.unk_F2[4] == 0) {
+                    this->unk_14C[i].unk_18 = (Rand_ZeroOne() * 3.0f) + 1.0f;
+                } else {
+                    this->unk_14C[i].unk_18 = (Rand_ZeroOne() * 3.0f) + 8.0f;
+                }
+                this->unk_14C[i].unk_1C++;
+                break;
+
+            case 1:
+                temp_f24 = globalCtx->view.eye.x + (spD0 * 120.0f);
+                temp_f28 = globalCtx->view.eye.y + (spCC * 120.0f);
+                temp_f30 = globalCtx->view.eye.z + (spC8 * 120.0f);
+
+                sqrt = sqrtf((f32)SQ(globalCtx->envCtx.windDir.x) + SQ(globalCtx->envCtx.windDir.y) +
+                             SQ(globalCtx->envCtx.windDir.z));
+                if (sqrt == 0.0f) {
+                    sqrt = 0.001f;
+                }
+                spC4 = -globalCtx->envCtx.windDir.x / sqrt;
+                spC0 = -globalCtx->envCtx.windDir.y / sqrt;
+                spBC = -globalCtx->envCtx.windDir.z / sqrt;
+
+                if (i == 0) {
+                    this->unk_144 += 0.049999997f * Rand_ZeroOne();
+                    this->unk_148 += 0.049999997f * Rand_ZeroOne();
```

Comment:
Can these be `1/20.0f` or something more plausible?

Reply:
Doesn't match unfortunately.